apps/src/bin/pravega_event_dumper.rs

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+
+use clap::Clap;
+
+use pravega_client::client_factory::ClientFactory;
+use pravega_client_shared::{Scope, Stream, ScopedStream};
+use pravega_video::event_serde::EventReader;
+use pravega_video::index::{IndexRecord, IndexRecordReader};
+use pravega_video::utils;
+use pravega_video::utils::{CurrentHead, SyncByteReader};
+use std::io::{ErrorKind, Read, Seek, SeekFrom};
+
+#[derive(Clap)]
+struct Opts {
+    /// Pravega controller in format "127.0.0.1:9090"
+    #[clap(short, long, default_value = "127.0.0.1:9090")]
+    controller: String,
+    /// Pravega scope
+    #[clap(long)]
+    scope: String,
+    /// Pravega stream
+    #[clap(long)]
+    stream: String,
+    /// Index number
+    #[clap(long, default_value = "10")]
+    index_num: u32,
+    #[clap(long)]
+    show_event: bool,
+    /// Pravega keycloak file
+    #[clap(long, default_value = "", setting(clap::ArgSettings::AllowEmptyValues))]
+    keycloak_file: String,
+}
+
+/// Demonstrate ability to write using the byte stream writer and read using the event reader.
```

Comment:
This comment might be misleading since `pravega_event_dumper` only dump events using event reader. Write using byte stream is not a part of this.

Reply:
Good point. Fix the comment.